users/miketronic/miketronic.c

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 @miketronic -- Mike B <mxb540@gmail.com>
+// SPDX-License-Identifier: GPL-2.0+
+
+#include "miketronic.h"
+
+// Add reconfigurable functions here, for keymap customization
+// This allows for a global, userspace functions, and continued
+// customization of the keymap.  Use _keymap instead of _user
+// functions in the keymaps
+
+
+// In keymaps, instead of writing _user functions, write _keymap functions
+// The __attribute__((weak)) allows for empty definitions here, and during
+// compilation, if these functions are defined elsewhere, they are written
+// over. This allows to include custom code from keymaps in the generic code
+// in this file.
+
+
+
+
+__attribute__ ((weak)) void matrix_init_keymap(void) { }
+__attribute__ ((weak)) void matrix_scan_keymap(void) { }
+__attribute__ ((weak)) void led_set_keymap(uint8_t usb_led) { }
+
+__attribute__ ((weak)) layer_state_t layer_state_set_keymap (layer_state_t state) {
+    return state;
+}
+// Init effect for RGB boards only
+#ifdef RGB_MATRIX_ENABLE
+void matrix_init_user(void) {
+    rgb_matrix_mode_noeeprom(RGB_MATRIX_SOLID_COLOR);
+}
+#endif
```

Comment:
If you're not calling the above functions,they won't ever get used. 
```suggestion
void matrix_init_user(void) {
#ifdef RGB_MATRIX_ENABLE
    rgb_matrix_mode_noeeprom(RGB_MATRIX_SOLID_COLOR);
#endif
    matrix_init_keymap();
}
```

Also, housekeeping_task_user, and keyboard_post_init_user are better options for the init and scan functions.

Reply:
I need to figure out the function stuff for the options you mentioned, hard to grasp these.  If this a show-stopper let me know.

Reply:
@drashna I read up on the quantum functions and updated matrix_init_user to keyboard_post_init_user.  I don't see a need for the housekeeping task though as I'm not concerned with turning off the rbg matrix on a time for instance, unless there is something else I could be doing with it?  Will commit/push after this is done compiling.